S4TF_Tutorial_5.ipynb

```diff
@@ -0,0 +1,1183 @@
h\nEm9qR8e2CN1AcKldy6ub16w2a0bDoSl9+75TwhEVti031CCNhZ88m7Hdbnn+/GuUqnwBXT6Jqrnr\nuiZYFkWBOx6zPwi85EmSMEiH9P2OXA1hNE2ubX3jj6M3EI0LdBxX3cByG8ZRwuEgG5ft2ByOGXXT\n0DWNaYFs1xuyLKdpW7wwIIxj6rKkrEq2h4y5UqDSUn2Hw4HseDQLz/N8phdzxbP36buO7W5nRKG1\n8EjbtjStYAq18rujGGNaMDlJEhM0NbFB44Q1p1uCnyAjtGWKxhDqkl8D33X52TWiMVqVJfvDnizP\nDEg/iROqQt6j6zigsktNL0ziWPx+zrJMY/NiifDG4XjkeBQ6qPZWNzRU1zXCvhoj2jQi6qyxqNIu\ns83vRBjEx3FcyqqiUuV313Vi2zweE8cRWLZxGdVqY9PpFMBgc7XNsB+IXbej2l0aC6vvRU3J1EMr\n/V40xCtNU4W+kEHRMcs4HA5vDaYCJe9nYykd0a8UqYA/QeDs+/4z4Ls/4edL4N/9V3xS/RwGrHr+\ns/Md3TA9HIso8CnKNXQN77/7LvP5nGQQsV58KfYNZcnxcGCXZWhqZ14U7JVw6WQ6oSwbMuWfbtk2\n2bEARG5rMBjw/Pm7pGnKYrEiDBOKIqeuWx49eoTneex2O4bDoRLQ6Oj7Dsd5m4Pv2A5lleO6HoEv\nC+bJk8dKyLjinXfewfM8NtsH4mTAe+++y8uXL8mOByoFQF4tl0ynU775Cx/w+7/7u6TDIbQNlm1x\nOB5pm4otW2bzGfOLOYc8Yzic4Ng2y+WSpm6E/aT85oejAce85PHjx8Sx2PHqdkNeyZR0NBmz+fyF\naGfmGa7nGR55OkixbJs3r16LKK6nNf9aAXHnmQFPa8Hmpml49Pgxu81WhlOLBXEck2U5lgOe7xOH\nkepJWSTDEaHrcnf3QOj7HIqc8Uh6llqZHyzK3cqU364KKH3bslqtiAcJi8WCPC9YblZsNluKouBb\nH36L6XjCfD7n9uYNV1fX3Ly5wbIswa8GAavViiiKjDqP7jEOBgOy7Z4gDKlV4Hn8WESly6LgX37y\nCbPJhDdv3oiaj++TDods1muBes2UI6V90pOsFSY2Ufqkw0HK7d2tIDSGIzpl7zufzcgVlbPve6Pf\neXN7y2QyIU1Teb9pyl6J9aZpSqmC7Hw+5+HhwbRWNBB/t98TqE3h4eEBgOlsxmwy4fPXX7Lb7Xjy\n+InxY9JaAkXhmTZDGEW4jmz+ibJZzvOc0Whk0CY62Glm0nA4NJbFh/2eQZrKWrQshumQDmGZbbYb\n0jQ1ZX4URaZyEIV+IVXoSiTPcyaTCZVqKeiyXJ/rrutYrpYCR+r+rFlngOhm9qes07EsLISF0PUy\n0XxrzNSLSGroOzhUZPmepippW4u7m1f89uYL9qtbttsHVku58XzPo+stmrbDtlAWDhaNMsfKioKq\nahgMh0rpvaLrwLKk1NOZRJZVhlc7HI5FpDYUbUB5kz/eLOmwLAgjH8uy6WlxPZvxWPpjQRByPOam\nWV4UGV97/jWlyNTQdTZlXTNQgwyAPMt5+vSpWDFsV3RNAz00bU1dihvgq1evKMuSb773LTJHPM3n\nF1OGwzGdBevVkkE6JElTaeAPR7ieZDZhGOKGYmGR5zlNq6a1KiM4ZR6iNOSelWldI83/wA+o6hMP\nWZdlg2SgZMBao06jp7ppIsDr3oKWzpRjtmo1uK6P33Y0fUdRluyV3zog1FYLDvuDMXgDmE6nuL5L\nWUo2HcShcNldl9VmzWq55J0nzwhcnzgIeXx9zWazMc8bKP75er0mPxxp25ap6nfevXpjMkXf9/Fc\nybrWmw1PnzwhjiLhc6upcd91xj8I2walmnTO+nFdl2JdKNmzzvQr/UA2pYNlmeRIw5D043xfnEi1\nza9ln+Tn9LpxFOJDS+xpX3tdTluua8Q7OsXkaZrmpCPrOli1ZXr6esBnpuhVBZ5r7pPJZKIyedfo\nAWj6qAbQn7dyQKwu9LS87d7m6Vu2TaMy+OFoRKjO8Xa7w/dPoi0/jnY4T7z0c+vvbdfB/hmanD8f\ngbPv6brm9L35AGK2Buelu3yLgiXZfSlA2jLDQZgTD/e3fLL8AoeS6rgjKw5SvkQS3GzLUllfR6++\nF7aDvAcZxBRUZcMhz1TpN8HGpW5q+s7F80KG6Zj5fM5wOFAfo/0JtC0JmpYNruMiNq89ti1sojCM\niMKE3U6UkbTqz3gy4rPPP+VwyHBdn+98598COu7vH1isVnJjNTXL5RLXk6mrZzl4gUvdNEZxaH84\n8Pzpe8zmFwyHqWqY21iOi+e6ZFnOaDJhv5OyW/eboiQijOXvN9utWTTDUcpeMa30oK6jx7IleNiO\nRa3EUNqzoYXmBetFXOTSt5qMJ1i2xWq1UotEBUpHO8dJ1dF2HQ3QVSW2qxTtsXB8jzCJ8WxX7DiU\n7UfV1JKNKMX19VbaHTKIcClLEbLomtbIxkHP3d0tvi/fj0big/XdX/wunu9xd3fPYvHAfr+nLCsW\niyVPnz2ja8XLaLPZKMk9+YxhGBp2z1HRF5fLpUEjxJ4EUF1invfETw6YJ3vlppUhiVb50VhTy7Le\nUkzqu45ePYetMkm9AQmpQcDovu9L1hbH7Hc7E3wdW2x6zz2wctW+6brOyOPpfqU2ZNOHnvY7rrB6\n2jPOeNO0Jsh6vo+vYEv6tSxONFHdNxUH0JJ0OJTM2PWM3mjfiRqXTmwCP6CLO0Nptv6IYHg+zKyq\nCtuy6f+seQ719CfFFdt+K1/rzgLmjw+Mur6nrzM6vyYMHaoa6Eqq4khd5Hg+9IjAadM0xJaD7XnY\nrgtFYZrZnqsWnSM7X1HWuG4AnU3Yd6qfJBPuyI44HHpj+JUkCZ4X0LYndevT++0kkFo9tu3Q9y1d\n14qZlQOuJRPGOI7pe4v7+3tlJZHTdT2r5
```

Comment:
The learning rate for the Adam optimizer might be a hair too low, when compared to the Python notebook. The defaults for the tf.keras Adam optimizer:

https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Adam

specify a learning rate of 0.001. The defaults for the Swift Adam optimizer look to be pretty much the same as for tf.keras, with the exception of epsilon:

https://github.com/tensorflow/swift-apis/blob/master/Sources/TensorFlow/Optimizer.swift

Reply:
I am getting these graphs for learning rate 0.001

![learningRate0 001](https://user-images.githubusercontent.com/31190928/59938057-9bf12c00-9471-11e9-8c20-858fdb938c6c.png)